src/components/date-picker/date-picker.tsx

```diff
@@ -98,123 +89,88 @@ export class DatePicker {
+        this.dateFormatter = new DateFormatter(this.language);
     }
 
     public componentWillLoad() {
-        switch (this.type) {
-            case 'date':
-                this.picker = new DateDatePicker(
-                    this.format,
-                    this.language,
-                    this.change
-                );
-                break;
-
-            case 'time':
-                this.picker = new TimePicker(
-                    this.format,
-                    this.language,
-                    this.change
-                );
-                break;
-
-            case 'week':
-                this.picker = new WeekPicker(
-                    this.format,
-                    this.language,
-                    this.change
-                );
-                break;
-
-            case 'month':
-                this.picker = new MonthPicker(
-                    this.format,
-                    this.language,
-                    this.change,
-                    translate
-                );
-                break;
-
-            case 'quarter':
-                this.picker = new QuarterPicker(
-                    this.format,
-                    this.language,
-                    this.change,
-                    translate
-                );
-                break;
-            case 'year':
-                this.picker = new YearPicker(
-                    this.format,
-                    this.language,
-                    this.change,
-                    translate
-                );
-                break;
-
-            case 'datetime':
-            default:
-                this.picker = new DatetimePicker(
-                    this.format,
-                    this.language,
-                    this.change
-                );
-                break;
-        }
+        this.formattedValue = this.dateFormatter.formatDateByType(this.value, this.type);
     }
 
     public componentDidLoad() {
         const textfield: HTMLElement = this.host.shadowRoot.querySelector(
             'limel-input-field'
         );
-        this.input = textfield.shadowRoot.querySelector('input');
-        this.container = this.host.shadowRoot.querySelector('.container');
+        const inputElement = textfield.shadowRoot.querySelector('input');
 
-        this.picker.init(this.input, this.container, this.value);
-        this.formattedValue = this.picker.formatDate(this.value);
-    }
+        let calendarElement: HTMLLimelDatePickerCalendarContainerElement;
+        calendarElement = document.querySelector(
+            `#${this.portalId} limel-date-picker-calendar-container`
+        )
 
-    public componentDidUnload() {
-        this.picker.destroy();
+        calendarElement.inputElement = inputElement;
+
+        setTimeout(() => {
+            this.showPortal = false;
+        }, 100); // Without this timeout value the width is set to zero on the picker in the webclient
```

Comment:
I would like to remove this comment but I also want people to understand why I have chosen the timeout value...

In the best of worlds I would want to completely get rid of the timeout function.

Reply:
Hmm… yeah, this didn't work with a timeout of 0 I guess?

Just in case you or anyone else reading this doesn't already know why I'm asking:
A timeout of zero is completely ok to use, because all that does is letting everything that's "behind" this function run before the timeout triggers. So there's no timing-issue with doing that (no pun intended). But if a non-zero timeout is necessary, that means we do have a timing-issue, and if the process we are waiting for happens to run too slow, we will be triggering the timeout too early.

Reply:
In lime-elements it worked fine but not in the webclient. Super confusing tbh. But I saw that there is another timeout in the datepickers as well which also has a value defined.

It seems like that was added to support changing the date directly in the input field, which we no longer support. (At least not in master). Maybe we could remove that timeout and then this timeout could be run without a value specified.

Reply:
⚡ 

I found a better solution which made it possible to get rid off the timeout value :)